rclcpp/src/rclcpp/node_interfaces/node_graph.cpp

```diff
@@ -381,26 +382,59 @@ convert_to_topic_info_list(const rcl_topic_endpoint_info_array_t & info_array)
   return topic_info_list;
 }
 
-template <const char * EndpointType, typename FunctionT>
+template<const char * EndpointType, typename FunctionT>
 static std::vector<rclcpp::TopicEndpointInfo>
 get_info_by_topic(
   rclcpp::node_interfaces::NodeBaseInterface * node_base,
   const std::string & topic_name,
   bool no_mangle,
   FunctionT rcl_get_info_by_topic)
 {
-  std::vector<rclcpp::TopicEndpointInfo> topic_info_list;
-
+  std::string fqdn;
   auto rcl_node_handle = node_base->get_rcl_node_handle();
-  auto fqdn = rclcpp::expand_topic_or_service_name(
-    topic_name,
-    rcl_node_get_name(rcl_node_handle),
-    rcl_node_get_namespace(rcl_node_handle),
-    false);    // false = not a service
+
+  if (no_mangle) {
+    fqdn = topic_name;
+  } else {
+    fqdn = rclcpp::expand_topic_or_service_name(
+      topic_name,
+      rcl_node_get_name(rcl_node_handle),
+      rcl_node_get_namespace(rcl_node_handle),
+      false);    // false = not a service
+
+    // Get the node options
+    const rcl_node_options_t * node_options = rcl_node_get_options(rcl_node_handle);
+    if (nullptr == node_options) {
+      throw std::runtime_error("Need valid node options in get_info_by_topic()");
+    }
+
+    // global before local so that local overwrites global
```

Comment:
nit: what is this comment referring to?

Reply:
I'm not sure what the comment is meant to say, but the way the remapping works is that there are "local" remappings in the `node_options->arguments` and "global" remappings in the `global_args`. But the global args are only used if `node_options->use_global_arguments` is true.

Reply:
I got the content of this comment from `rclcpp`'s `node_parameters.cpp`. I didn't notice that the context is very different. I'll remove this comment.